src/components/pages/ImageClassification/ImageClassificationTable/ImageClassificationContainer.js

```diff
@@ -38,11 +48,17 @@ const ImageClassificationContainer = (props) => {
 
         setImages(response.results)
 
-        const areAllImagesProcessed =
+        // if an upload is slow, this may return true before
+        // the upload is complete if previously processed images are present
+        const areAllUploadedImagesProcessed =
           response.results.length &&
           response.results.every((file) => getIsImageProcessed(file.classification_status?.status))
 
-        if (areAllImagesProcessed) {
+        // ... so we need to check if any images are still being uploaded as well
+        // (since useState values will be stale within a closure/setInterval callback,
+        // we use a reference to access the current value for isUploading instead of the stale closure value directly)
```

Comment:
this stale state thing is confusing so I am leaning on overdoing redundant comments for the hacky fix.